projects/common/models/slider-element.ts

```diff
@@ -0,0 +1,31 @@
+import {FontElement, SurfaceUIElement} from '../interfaces/UIElementInterfaces';
+import { InputElement, UIElement} from './uI-element';
+import {initFontElement, initSurfaceElement} from '../util/unit-interface-initializer';
+
+export class SliderElement extends InputElement implements FontElement, SurfaceUIElement {
+  minValue: number = 0;
+  maxValue: number = 100;
+  showValues: boolean = true;
+
+  bold: boolean = false;
+  font: string = 'Roboto';
+  fontColor: string = 'black';
+  fontSize: number = 18;
+  italic: boolean = false;
+  lineHeight: number = 120;
+  underline: boolean = false;
+
+  backgroundColor: string = '#AAA0';
+
+  constructor(serializedElement: UIElement) {
+    super(serializedElement);
+    Object.assign(this, serializedElement);
+    Object.assign(this, initFontElement(serializedElement));
+    if (!serializedElement.backgroundColor) {
+      serializedElement.backgroundColor = '#d3d3d300';
+    }
+    Object.assign(this, initSurfaceElement(serializedElement));
+    // todo: delete this.label --> label must be declared as optional
```

Comment:
Wir haben gerade über die Optionalität von Labels diskutiert und beschlossen, dass wir gerne erzwingen möchten dass alle Inputelemente ein Label implementieren. Dieses kann gerne leer sein und damit nicht erscheinen.

Reply:
Dann muss die Verwendung von Label geklärt werden:
* Wenn Ihr das intern haben wollt, dann sollte es 'controlLabel' oder 'name' heißen.
* Wenn das eine Instruktion sein soll, dann sollte es 'prompt' heißen.
* Wenn es einfach nur eine Überschrift ist, dann sollte es 'title' heißen.
* usw.

Das Problem entsteht ja vor allem im Property-Panel, wo die Verwendung durch die Beschriftung des Eingabefeldes für label deutlich werden sollte. Man kann hier aber nur eine Beschriftung vorsehen.

Im Element für die Wort-Korrektur gibt es ein buttonLabel, wo ich eigentlich auch das ursprüngliche label nehmen könnte. Aber es gibt auch ein Eingabefeld, das kein Label haben soll. Also was ist dann 'Label'?

Reply:
Man könnte eine ReadOnly-Property haben, die man in jedem Element überschreiben muss, und die liefert eine Bezeichnung, die als Label in der Editor-UI (und auch z. B. im Kodierschema als Variablen-Label) verwendet werden kann. Wo diese Bezeichnung herkommt, ist offen: Titel, Prompt, Button-Label oder was auch immer, das entscheidet der Entwickler der Componente.